crypto/asn1/asn1_lib.c

```diff
@@ -279,6 +279,10 @@ int ASN1_STRING_set(ASN1_STRING *str, const void *_data, int len)
         else
             len = strlen(data);
     }
+
+    if ( len < 0)
+        return 0;
```

Comment:
Description is correct, but solution does not formally align with description. Just imagine 2^32+n long string on 64-bit system. Well, even on 32-bit it's formally incorrect, because conversion from unsigned to signed with most significant bit is not formally defined. Correct solution would be to declare size_t variable that would receive strlen return value, check that for boundary condition, and only then assign it to len.